ROS2 branch is failing

**Description**

The `ros2` branch is failing on ros2 master builds. See output below, looks like its related to gtest and depreciated feature use. CI job in Nav2 validates: https://app.circleci.com/pipelines/github/ros-planning/navigation2/4053/workflows/2a54ca7b-1b38-4182-b45e-b4a638d20760/jobs/15522

**To Reproduce**

Build `ros2` based on `ros2 master` 

**Expected behavior**

Builds correctly :-)
**Environment (please complete the following information):**

 - OS: Ubuntu 20
 - Gazebo version: N/A
 - ROS version: ROS2 master
  - gazebo_ros_pkgs: Source of ros2 branch
 - glx server in use: N/A

**Additional context**

```
]0;colcon build [18/21 done] [0 ongoing]Starting >>> gazebo_ros
--- stderr: gazebo_ros
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:38: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:73: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                                                         ^
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:73: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                                                         ^
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp: In function ‘std::string gtest_PluginsTestPlugins_EvalGenerateName_(const testing::TestParamInfo<TestParams>&)’:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:472:43: error: expected primary-expression before ‘)’ token
  472 |       auto t = std::make_tuple(__VA_ARGS__);                                  \
      |                                           ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:473:48: error: template argument 1 is invalid
  473 |       static_assert(std::tuple_size<decltype(t)>::value <= 2,                 \
      |                                                ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:479:23: error: expected primary-expression before ‘)’ token
  479 |         DUMMY_PARAM_))))(info);                                               \
      |                       ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:181,
                 from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-param-util.h: In instantiation of ‘void testing::internal::TestNotEmpty() [with T = int]’:
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1:   required from here
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-param-util.h:382:27: error: static assertion failed: Empty arguments are not allowed.
  382 |   static_assert(sizeof(T) == 0, "Empty arguments are not allowed.");
      |                 ~~~~~~~~~~^~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-inconsistent-missing-override’
make[2]: *** [test/CMakeFiles/test_plugins.dir/build.make:63: test/CMakeFiles/test_plugins.dir/test_plugins.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:530: test/CMakeFiles/test_plugins.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< gazebo_ros [23.7s, exited with code 2]
]0;colcon build [18/21 done] [1 ongoing]]0;colcon build [19/21 done] [0 ongoing]
Summary: 18 packages finished [1min 44s]
  1 package failed: gazebo_ros
  2 packages had stderr output: behaviortree_cpp_v3 gazebo_ros
  2 packages not processed
```

--- a/gazebo_plugins/test/test_gazebo_ros_camera.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_camera.cpp
@@ -81,7 +81,7 @@ TEST_P(GazeboRosCameraTest, CameraSubscribeTest)
   sub.shutdown();
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosCamera, GazeboRosCameraTest, ::testing::Values(
     TestParams({"worlds/gazebo_ros_camera.world", "test_cam/camera/image_test"}),
     TestParams({"worlds/gazebo_ros_camera_16bit.world", "test_cam_16bit/image_test_16bit"})

--- a/gazebo_plugins/test/test_gazebo_ros_camera_distortion.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_camera_distortion.cpp
@@ -208,7 +208,7 @@ TEST_P(GazeboRosCameraDistortionTest, CameraSubscribeTest)
   }
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosCameraDistortion, GazeboRosCameraDistortionTest, ::testing::Values(
     TestParams(
       {"worlds/gazebo_ros_camera_distortion_barrel.world",

--- a/gazebo_plugins/test/test_gazebo_ros_depth_camera.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_depth_camera.cpp
@@ -136,7 +136,7 @@ TEST_P(GazeboRosDepthCameraTest, DepthCameraSubscribeTest)
   EXPECT_EQ(8.0, image_stamp_pcl.sec);
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosDepthCamera, GazeboRosDepthCameraTest, ::testing::Values(
     TestParams(
       {"worlds/gazebo_ros_depth_camera.world",

--- a/gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp
@@ -122,12 +122,12 @@ TEST_P(GazeboRosDiffDriveTest, Publishing)
   EXPECT_NEAR(0.1, vehicle->WorldAngularVel().Z(), tol);
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosDiffDrive, GazeboRosDiffDriveTest, ::testing::Values(
     TestParams({"worlds/gazebo_ros_diff_drive.world"}),
     TestParams({"worlds/gazebo_ros_skid_steer_drive.world"})
     // cppcheck-suppress syntaxError
-  ), );
+  ));
 
 int main(int argc, char ** argv)
 {

--- a/gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp
@@ -127,7 +127,7 @@ INSTANTIATE_TEST_SUITE_P(
     TestParams({"worlds/gazebo_ros_diff_drive.world"}),
     TestParams({"worlds/gazebo_ros_skid_steer_drive.world"})
     // cppcheck-suppress syntaxError
-  ));
+));
 
 int main(int argc, char ** argv)
 {

--- a/gazebo_plugins/test/test_gazebo_ros_multi_camera.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_multi_camera.cpp
@@ -106,7 +106,7 @@ TEST_P(GazeboRosMultiCameraTest, CameraSubscribeTest)
   sub_right.shutdown();
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosMultiCamera, GazeboRosMultiCameraTest, ::testing::Values(
     TestParams(
       {"worlds/gazebo_ros_multi_camera.world",

--- a/gazebo_plugins/test/test_gazebo_ros_multicamera.cpp
+++ b/gazebo_plugins/test/test_gazebo_ros_multicamera.cpp
@@ -106,7 +106,7 @@ TEST_P(GazeboRosMultiCameraTest, CameraSubscribeTest)
   sub_right.shutdown();
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosMultiCamera, GazeboRosMultiCameraTest, ::testing::Values(
     TestParams(
       {"worlds/gazebo_ros_multicamera.world",

--- a/gazebo_ros/test/test_plugins.cpp
+++ b/gazebo_ros/test/test_plugins.cpp
@@ -67,7 +67,7 @@ TEST_P(TestPlugins, TestTopicsReceived)
   rclcpp::sleep_for(1s);
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   Plugins, TestPlugins, ::testing::Values(
     TestParams({{"-s", "./libargs_init.so"}, {"test"}}),
     TestParams({{"-s", "./libcreate_node_without_init.so"}, {"test"}}),
@@ -80,7 +80,7 @@ INSTANTIATE_TEST_CASE_P(
         "worlds/ros_world_plugin.world"}, {"test"}}),
     TestParams({{"-s", "libgazebo_ros_init.so", "worlds/sdf_node_plugin.world"}, {"/foo/my_topic"}})
     // cppcheck-suppress syntaxError
-  ), );
+  ));
 
 int main(int argc, char ** argv)
 {

--- a/gazebo_ros/test/test_plugins.cpp
+++ b/gazebo_ros/test/test_plugins.cpp
@@ -80,7 +80,7 @@ INSTANTIATE_TEST_SUITE_P(
         "worlds/ros_world_plugin.world"}, {"test"}}),
     TestParams({{"-s", "libgazebo_ros_init.so", "worlds/sdf_node_plugin.world"}, {"/foo/my_topic"}})
     // cppcheck-suppress syntaxError
-  ));
+));
 
 int main(int argc, char ** argv)
 {

